### ISSUE TITLE
listAlbums returns empty

### ISSUE THREAD

**Issue by duzenko** [Firelands128/photo_gallery#37]
```
    videoAlbums = await PhotoGallery.listAlbums(mediumType: MediumType.video);
```
returns one album with 0 length even though there are videos on device

**Comment by Firelands128** [comment]
Please provide more info to reproduce this issue.

**Comment by duzenko** [comment]
I think it might have something to do with permissions
The photo_manager plugin I used instead of yours shows the permission request properly and works afterwards

**Comment by mjmon** [comment]
> I think it might have something to do with permissions The photo_manager plugin I used instead of yours shows the permission request properly and works afterwards

I agree, the limited state should be handled as well explicitly on the UI.
Maybe photo_manager should be included on the docu

**Comment by unclemeric** [comment]
修改Podfile文件：
post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
    # 添加以下代码可解决问题
    target.build_configurations.each do |config|
      config.build_settings['GCC_PREPROCESSOR_DEFINITIONS'] ||= [
        '$(inherited)',
        ## dart: PermissionGroup.camera
        'PERMISSION_CAMERA=1',
        # dart: PermissionGroup. photos
        'PERMISSION_PHOTOS=1',
        'PERMISSION_PHOTOS_ADD_ONLY=1',
        # dart: [PermissionGroup. location, PermissionGroup. locationAlways, PermissionGroup. locationWhenInUse]
        # 'PERMISSION_LOCATION=1',
        # dart: PermissionGroup.mediaLibrary
        'PERMISSION MEDIA LIBRARY=1'
      ]
    end
  end
end
